app/src/main/java/org/mifos/selfserviceapp/presenters/ThirdPartyTransferPresenter.java

```diff
@@ -0,0 +1,191 @@
+package org.mifos.selfserviceapp.presenters;
+
+import android.content.Context;
+
+import org.mifos.selfserviceapp.R;
+import org.mifos.selfserviceapp.api.DataManager;
+import org.mifos.selfserviceapp.injection.ActivityContext;
+import org.mifos.selfserviceapp.models.beneficary.Beneficiary;
+import org.mifos.selfserviceapp.models.payload.TransferPayload;
+import org.mifos.selfserviceapp.models.templates.account.AccountOption;
+import org.mifos.selfserviceapp.models.templates.account.AccountOptionsTemplate;
+import org.mifos.selfserviceapp.presenters.base.BasePresenter;
+import org.mifos.selfserviceapp.ui.views.ThirdPartyTransferView;
+import org.mifos.selfserviceapp.utils.MFErrorParser;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import okhttp3.ResponseBody;
+import rx.Observable;
+import rx.Subscriber;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.functions.Action1;
+import rx.functions.Func1;
+import rx.schedulers.Schedulers;
+import rx.subscriptions.CompositeSubscription;
+
+/**
+ * Created by dilpreet on 21/6/17.
+ */
+
+public class ThirdPartyTransferPresenter extends BasePresenter<ThirdPartyTransferView> {
+
+    private DataManager dataManager;
+    private CompositeSubscription subscription;
+
+    @Inject
+    public ThirdPartyTransferPresenter(DataManager dataManager, @ActivityContext Context context) {
+        super(context);
+        this.dataManager = dataManager;
+        subscription = new CompositeSubscription();
+    }
+
+    @Override
+    public void attachView(ThirdPartyTransferView mvpView) {
+        super.attachView(mvpView);
+    }
+
+    @Override
+    public void detachView() {
+        super.detachView();
+        subscription.clear();
+    }
+
+    public void loadTransferTemplate() {
```

Comment:
@dilpreet96 make rxjava zip request to combine the loadTransferTemplate and loadBeneficiaries request.
because if anyone will fail user can't make the transfer.